File: 백준/Silver/7568. 덩치/덩치.java
import java.io.*;
import java.util.*;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int count = Integer.parseInt(br.readLine());
        int[][] arr = new int[count][2];
        for(int i =0; i<count; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            int weight = Integer.parseInt(st.nextToken());
            int height = Integer.parseInt(st.nextToken());
            arr[i][0] = weight;
            arr[i][1] = height;
        }
        for(int i =0; i<count; i++) {
            int rank = 1;
            for(int j=0; j<count; j++) {
                if(i==j) continue;
                if(arr[i][0] < arr[j][0] && arr[i][1] < arr[j][1]) {
                    rank++;
                }
            }
            System.out.print(rank + " ");
        }
    }
}

File: 백준/Silver/7568. 덩치/덩치.kt
import java.util.StringTokenizer

fun main() {
    System.`in`.bufferedReader().use {
        val count = it.readLine().toInt()
        val array = Array(count) { IntArray(2) }
        for (i in 0 until count) {
            val st = StringTokenizer(it.readLine())
            val weight = st.nextToken().toInt()
            val height = st.nextToken().toInt()
            array[i][0] = weight
            array[i][1] = height
        }
        for(i in 0 until count) {
            var rank = 1
            for(j in 0 until count) {
                if (i == j) continue
                if (array[i][0] < array[j][0] && array[i][1] < array[j][1]) {
                    rank++
                }
            }
            print("$rank ")
        }
    }
}